examples/identity_hasher.rs

```diff
@@ -0,0 +1,77 @@
+use multihash::derive::Multihash;
+use multihash::{Error, Hasher, MultihashDigest, MultihashGeneric, Sha2_256};
+
+/// update appends to end of buffer but truncates/ignores bytes after len > S
+#[derive(Debug)]
+pub struct IdentityTrunk<const S: usize> {
+    cursor: usize,
+    arr: [u8; S],
+}
+
+impl<const S: usize> Default for IdentityTrunk<S> {
+    fn default() -> Self {
+        Self {
+            cursor: 0,
+            arr: [0u8; S],
+        }
+    }
+}
+impl<const S: usize> Hasher for IdentityTrunk<S> {
+    fn update(&mut self, input: &[u8]) {
+        let src_end = (self.cursor + input.len()).min(self.arr.len());
+        let input_end = input.len().min(self.arr.len() - self.cursor);
+
+        self.arr[self.cursor..src_end].copy_from_slice(&input[..input_end]);
+        self.cursor = src_end;
+    }
+    fn finalize(&mut self) -> &[u8] {
+        &self.arr[..self.cursor]
+    }
+    fn reset(&mut self) {
+        *self = Self {
+            cursor: 0,
+            arr: [0u8; S],
+        };
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, Multihash, PartialEq)]
+#[mh(alloc_size = 64)]
+pub enum Code {
+    #[mh(code = 0x00, hasher = IdentityTrunk::<64>)]
+    IdentityTrunk,
+    #[mh(code = 0x12, hasher = Sha2_256)]
+    Sha2_256,
+}
+
+#[allow(unused)]
```

Comment:
Instead of `#[allow(unused)]` you could add asserts. This way one would also get an idea what the output of those functions look like.